ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -182,10 +183,20 @@ public boolean accept(Path path) {
   public static final int MAX_STATEMENTS_PER_TXN = 10000;
   public static final Pattern LEGACY_BUCKET_DIGIT_PATTERN = Pattern.compile("^[0-9]{6}");
   public static final Pattern BUCKET_PATTERN = Pattern.compile("bucket_([0-9]+)(_[0-9]+)?$");
+  private static final Set<Integer> READ_TXN_TOKENS = new HashSet<Integer>();
 
   private static Cache<String, DirInfoValue> dirCache;
   private static AtomicBoolean dirCacheInited = new AtomicBoolean();
 
+  static {
+    READ_TXN_TOKENS.addAll(Arrays.asList(
```

Comment:
does it cover SHOW LOCKS and EXPLAIN LOCKS?

Reply:
Yes, they are covered